### PR TITLE
Update Gradle to 6.2

### PR DESCRIPTION
**Pull request by SR-Lut3t1um**

### CODE REVIEW

**Comment by gastaldi** [comment]
Weird, Gradle 6.2 reports #7249 as `BUILD SUCCESSFUL` after the failure, need to investigate why

**Comment by gastaldi** [comment]
Hm, same with 6.0.1, seems to be unrelated

**Comment by SR-Lut3t1um** [comment]
Yes I'm pretty sure it is. Just wanted to make sure and does no harm either.

**Comment by gastaldi** [comment]
You also need to update https://github.com/quarkusio/quarkus/blob/49ea44edc9e591ce6681013835f9d230b1646958/independent-projects/tools/pom.xml#L30 if you want Gradle 6.2 to be used in new projects

**Comment by gastaldi** [comment]
BTW the following files may also be modified:

```
	modified:   gradle/wrapper/gradle-wrapper.jar
	modified:   gradle/wrapper/gradle-wrapper.properties
	modified:   gradlew
	modified:   gradlew.bat
```

**Comment by gastaldi** [comment]
code.quarkus.io does not use `gradle wrapper` and has the wrapper sources here: https://github.com/quarkusio/code.quarkus.io/tree/master/src/main/resources/creator/gradlew

**Comment by SR-Lut3t1um** [comment]
ah ok makes sense. Gonne fix now

**Comment by gastaldi** [comment]
Are you sure the files in https://github.com/quarkusio/quarkus/tree/3c4885f943622b28283ee4a44be80e677dc2f84b/devtools/gradle/gradle/wrapper + the gradlew scripts don't change? 

Also can you squash your commits in a single commit for this PR?

**Comment by SR-Lut3t1um** [comment]
1. I'm also going to make a PR there to bump the gradle version.
2. Sure

**Comment by SR-Lut3t1um** [comment]
I did 1.

**Comment by gastaldi** [comment]
There is also https://github.com/quarkusio/quarkus/blob/13c28058423690b17bcac384c9433a27f50aac52/build-parent/pom.xml#L58 which needs to be updated

**Comment by SR-Lut3t1um** [comment]
So I changed the last file that needed change and also crated the final pull request.

https://github.com/quarkusio/code.quarkus.io/pull/298

**Comment by SR-Lut3t1um** [comment]
I guess the pipeline crashed...

**Comment by SR-Lut3t1um** [comment]
Can this PR be accepted?

**Review by gastaldi — DISMISSED** [review]

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java [hunk 1]
```diff
@@ -269,7 +267,7 @@ public void startDev() {
             final DependencySet compileCpDependencies = compileCp.getAllDependencies();
 
             for (Dependency dependency : compileCpDependencies) {
-                if (!(dependency instanceof ProjectDependency)) {
+                if (dependency == null) {
```
Comment (aloubyansky) [inline devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java]: Revert this change, please.

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java [hunk 2]
```diff
@@ -344,9 +342,8 @@ public void startDev() {
             }
 
             for (AppDependency appDependency : appModel.getFullDeploymentDeps()) {
-                if (!projectDependencies.contains(appDependency.getArtifact().getKey())) {
-                    addToClassPaths(classPathManifest, context, appDependency.getArtifact().getPath().toFile());
-                }
+                appDependency.getArtifact().getKey();
```
Comment (aloubyansky) [inline devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java]: Revert this change please.

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/unused/RemoveUnusedBeansTest.java [hunk 3]
```diff
@@ -190,7 +190,7 @@ BigInteger unusedNumber() {
     static class UsedViaInstanceWithUnusedProducer {
 
         @Produces
-        Long unusedLong = new Long(0);
+        Long unusedLong = 0L;
```
Comment (aloubyansky) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/unused/RemoveUnusedBeansTest.java]: Revert this change please.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/CuratedApplication.java [hunk 4]
```diff
@@ -106,13 +106,13 @@ public AugmentAction createAugmentor() {
...
      * {@code Function<Map<String, Object>, List<Consumer<BuildChainBuilder>>>}
-     * which is used to generate a list of build chain customisers to control the build.
+     * which is used to generate a list of build chain customizers to control the build.
      */
     public AugmentAction createAugmentor(String functionName, Map<String, Object> props) {
         try {
             Class<?> augmentor = getAugmentClassLoader().loadClass(AUGMENTOR);
             Function<Object, List<?>> function = (Function<Object, List<?>>) getAugmentClassLoader().loadClass(functionName)
-                    .newInstance();
+                    .getDeclaredConstructor().newInstance();
```
Comment (aloubyansky) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/CuratedApplication.java]: Revert this change, please.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/CuratedApplication.java [hunk 5]
```diff
@@ -127,7 +127,7 @@ private Object runInCl(String consumerName, Map<String, Object> params, QuarkusC
             Thread.currentThread().setContextClassLoader(cl);
             Class<? extends BiConsumer<CuratedApplication, Map<String, Object>>> clazz = (Class<? extends BiConsumer<CuratedApplication, Map<String, Object>>>) cl
                     .loadClass(consumerName);
-            BiConsumer<CuratedApplication, Map<String, Object>> biConsumer = clazz.newInstance();
+            BiConsumer<CuratedApplication, Map<String, Object>> biConsumer = clazz.getDeclaredConstructor().newInstance();
```
Comment (aloubyansky) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/CuratedApplication.java]: Revert this change, please.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapWagonProvider.java [hunk 6]
```diff
@@ -34,11 +34,11 @@ public Wagon lookup(String roleHint) throws Exception {
         final Class<?> cls = loadClass(impl, roleHint);
         final Object wagon;
         try {
-            wagon = cls.newInstance();
+            wagon = cls.getDeclaredConstructor().newInstance();
```
Comment (aloubyansky) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapWagonProvider.java]: Revert this change, please.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapWagonProvider.java [hunk 7]
```diff
@@ -34,11 +34,11 @@ public Wagon lookup(String roleHint) throws Exception {
...
         } catch (Throwable t) {
             throw new IllegalStateException("Failed to instantiate Wagon impl " + impl, t);
         }
-        return Wagon.class.cast(wagon);
+        return (Wagon) wagon;
```
Comment (aloubyansky) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapWagonProvider.java]: Revert this change, please.

**Review by aloubyansky — CHANGES_REQUESTED** [review]
Please revert everything except for the gradle version upgrade. Thanks!

**Comment by SR-Lut3t1um** [comment]
I did.

**Review by aloubyansky — APPROVED** [review]
Thanks!

**Comment by aloubyansky** [comment]
BTW, if you do open a PR with those kind of changes, please, let me know. Thanks.

**Comment by gsmet** [comment]
This PR needs a rebase unfortunately.

Once it's done, @gastaldi can you validate everything is OK for you?

**Comment by gastaldi** [comment]
@SR-Lut3t1um you need to rebase your branch: 

- `git rebase master`
- `git push -f`

**Comment by gastaldi** [comment]
Seems that there are conflicts. Can you `git pull` your master branch and rebase again?

**Comment by gsmet** [comment]
I rebased and fixed the conflicts.

**Comment by gastaldi** [comment]
Thanks! I tested locally and the Gradle build works great

**Pull request merged**